src/main/java/executor/api/controller/ProxySourceController.java

```
@@ -0,0 +1,36 @@
+package executor.api.controller;
+
+import executor.api.model.ProxyConfigHolderDTO;
+import executor.api.service.impl.ProxySourceQueueHandler;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Optional;
+
+@RestController
+@RequestMapping(value = "/api/v1")
+
+public class ProxySourceController {
+
+    private final ProxySourceQueueHandler proxySourceQueueHandler;
+
+    public ProxySourceController(ProxySourceQueueHandler proxySourceQueueHandler) {
+        this.proxySourceQueueHandler = proxySourceQueueHandler;
+    }
+
+    @PostMapping(value = "/proxy/add")
+    public ResponseEntity<?> addProxy(@RequestBody ProxyConfigHolderDTO proxyConfigHolderDTO) {
+        proxySourceQueueHandler.add(proxyConfigHolderDTO);
+        return ResponseEntity.ok("Added successfully");
+    }
+
+    @GetMapping(value = "/proxy/get")
+    public ProxyConfigHolderDTO getProxy() {
+        Optional<ProxyConfigHolderDTO> optional = proxySourceQueueHandler.poll();
+        return optional.get();
```

get() is not safe without checking.
We can throw an exception here with 404 or ..

---

Sure, thanks for the advice!